libs/domain/cart/entries/src/entries.component.ts

```diff
@@ -1,169 +1,168 @@
   );
 
-  protected onUpdate(e: CustomEvent, { groupKey, sku }: CartEntry): void {
-    this.currentlyUpdated$.next(groupKey);
-    this.cartService.updateEntry({
-      groupKey,
-      quantity: e.detail.quantity,
-    });
+  connectedCallback(): void {
+    this.addEventListener('submit', this.onSubmit as EventListener);
+    super.connectedCallback();
   }
 
-  protected onRemove({ groupKey }: CartEntry): void {
-    this.cartService.deleteEntry({ groupKey });
-  }
+  // TODO: implement loading state
+  protected override render(): TemplateResult | void {
+    if (this.isEmpty) {
+      return this.renderEmpty();
+    }
 
-  protected renderEmpty(): TemplateResult {
     return html`
-      <section>
-        <oryx-icon type="cart"></oryx-icon>
-        <p>Your shopping cart is empty</p>
-        <oryx-button>
-          <button>Shop now</button>
-        </oryx-button>
-      </section>
+      <oryx-heading>
+        <h1>
+          ${i18n('cart.totals.<count>-items', {
+            count: this.totalQuantity,
+          })}
+        </h1>
+      </oryx-heading>
+
+      ${repeat(
+        this.entries ?? [],
+        (entry) => entry.groupKey,
+        (entry) =>
+          html`
+            <oryx-cart-entry
+              .key=${entry.groupKey}
+              ?readonly=${this.componentOptions.readonly}
+            ></oryx-cart-entry>
+          `
+      )}
+      ${when(this.removeGroupKey, () =>
+        this.renderConfirmation(this.removeGroupKey)
+      )}
     `;
   }
 
-  protected renderEntries(entriesData: EntriesData): TemplateResult {
-    const { entries, loading, currentlyUpdated, options } = entriesData;
-
-    return html` ${repeat(
-      entries,
-      (entry) => entry.groupKey,
-      (entry) => html`
-        <cart-entry
-          ?inert=${loading}
-          .options=${{
-            ...options,
-            ...entry,
-            disabled: loading,
-            updating: entry.groupKey === currentlyUpdated,
-          }}
-          @submit=${(e: CustomEvent): void => this.onUpdate(e, entry)}
-          @oryx.remove=${(): void => this.onRemove(entry)}
-        ></cart-entry>
-      `
-    )}`;
-  }
-
-  protected renderItemsCount(entriesData: EntriesData): TemplateResult {
-    const { options, entries } = entriesData;
-
-    if (options.hideItemsCount) {
-      return html``;
+  /**
+   * Handles updates on the entry. When the quantity is 0, the entry is going
+   * to be removed unless the component options require to seek for confirmation first.
+   */
+  protected onSubmit(ev: CustomEvent<CartEntryChangeEventDetail>): void {
+    const { groupKey, quantity } = ev.detail;
+
+    if (quantity === 0) {
+      if (this.componentOptions?.silentRemove) {
+        this.removeEntry(groupKey);
+      } else {
+        this.removeGroupKey = groupKey;
+      }
+    } else {
+      this.cartService.updateEntry({ groupKey, quantity }).subscribe({
+        complete: () => {
+          if (this.componentOptions.notifyOnUpdate) {
```

Comment:
OK, to keep it. 
Just wondering if we need this logic in this component. 
We don't have global configuration for this, so basically, if you'll have more compnents that will update something in the cart, each of them should have this option. 

The same, if we would like to move removing to entry component (wihch I wonder, shouldn't be a case, then entries would just enumerate through every entry, but I guess you did consider it).

Reply:
covered in https://spryker.atlassian.net/browse/HRZ-2532